### PR TITLE
Support for benchmarking editor code.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
Sometimes it's hard to get concrete numbers for the performance of some code we've written especially for the internals of the editor which aren't directly tied to the rendering.

**Fix:**
Created a benchmarking "application" that can call into editor code and run it to see what kind of performance we get, which uses the `benny` library to do the heavy lifting and reporting. Now we can create alternative implementations of important functions and then compare and contrast their performance.

**Screenshot:**
image

**Commit Details:**
- Added `benny` dev dependency.
- Added `build-benchmark` and `benchmark` scripts to `editor/package.json`.
- Moved some functions and a const from `uid-utils.ts` to `element-path.ts` because they caused a circular reference.
- Created `fake-localforage.ts` so that it can be used to shim in inplace of the real `localforage` because that does not like being run by Node.
- Implemented `benchmarkBuildTree` to test this and because it will likely be changed imminently so we can use this as a comparison point.
- Added `benchmarks.benchmark.ts` as the root of all the benchmarking functions.
- Created new webpack config to launch the benchmarking.

### CODE REVIEW

File: editor/src/core/shared/element-path.ts [hunk 1]
```diff
@@ -8,7 +8,6 @@ import {
 import { arrayEquals, longestCommonArray, identity, fastForEach } from './utils'
 import { replaceAll } from './string-utils'
 import { last, dropLastN, drop, splitAt, flattenArray, dropLast } from './array-utils'
-import { extractOriginalUidFromIndexedUid } from './uid-utils'
```
Comment (balazsbajorics) [inline editor/src/core/shared/element-path.ts:11]: was this change needed to make it compile? or just an improvement related to your example test case?

Comment (seanparsons) [reply editor/src/core/shared/element-path.ts:11]: It was a change needed to avoid a circular reference.

Comment (balazsbajorics) [reply editor/src/core/shared/element-path.ts:11]: oh boy

**Review by ruggi — APPROVED** [review]

**Review by balazsbajorics — APPROVED** [review]
nice!!

File: editor/src/core/shared/element-path.ts (hunk 1 above)
Comment (Rheeseyb) [reply editor/src/core/shared/element-path.ts:11]: So this would be the circular reference that has bitten us a number of times in tests!

File: editor/src/utils/fake-localforage.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class FakeLocalForage {
```
Comment (Rheeseyb) [inline editor/src/utils/fake-localforage.ts:1]: We do something similar to this via a jest mock, except that backs it with an object to actually store and retrieve the results. Maybe it's worth doing that to prevent confusing issues later down the line? <https://github.com/concrete-utopia/utopia/blob/2a66237225cf36e7f6cf259010e13d03316dacfa/editor/src/components/editor/persistence/persistence.spec.ts#L154-L167>

**Review by Rheeseyb — APPROVED** [review]
Very nice!

File: editor/src/utils/fake-localforage.ts (hunk 2 above)
Comment (seanparsons) [reply editor/src/utils/fake-localforage.ts:1]: That code was the inspiration for mine, for benchmarking I think we don't want it to do anything because it could skew results in unexpected ways.
